### PR TITLE
atuin: set broken for aarch64-darwin

### PR DESCRIPTION
**Pull request by ixnij**
I cannot successfully build this package on the aarch64-darwin platform, and the upstream does not seem to support aarch64-darwin.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/atuin/default.nix [hunk 1]
```diff
@@ -27,5 +27,6 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/ellie/atuin";
     license = licenses.mit;
     maintainers = [ maintainers.onsails ];
+    badPlatforms = [ "aarch64-darwin" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/atuin/default.nix:30]:
```suggestion
    broken = [ "aarch64-darwin" ];
```

What is the error you get?

Comment (ixnij) [reply pkgs/tools/misc/atuin/default.nix:30]: https://github.com/ellie/atuin/issues/220#issue-1071354928

Comment (SuperSandro2000) [reply pkgs/tools/misc/atuin/default.nix]: I would guess that libsodium-sys does not support darwin-arach64.

File: pkgs/tools/misc/atuin/default.nix [hunk 2]
```diff
@@ -27,5 +27,6 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/ellie/atuin";
     license = licenses.mit;
     maintainers = [ maintainers.onsails ];
+    broken = [ "aarch64-darwin" ];
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/atuin/default.nix:30]:
```suggestion
    broken = stdenv.isDarwin && stdenv.isAarch64
```
my bad

File: pkgs/tools/misc/atuin/default.nix [hunk 3]
```diff
@@ -27,5 +27,6 @@ rustPlatform.buildRustPackage rec {
     homepage = "https://github.com/ellie/atuin";
     license = licenses.mit;
     maintainers = [ maintainers.onsails ];
+    broken = stdenv.isDarwin && stdenv.isAarch64
```
Comment (SuperSandro2000) [inline pkgs/tools/misc/atuin/default.nix:30]:
```suggestion
    broken = stdenv.isDarwin && stdenv.isAarch64;
```

**Comment by ellie** [comment]
Hey all

I've confirmed via three M1 macbooks, at least two of which are running Monterey, that Atuin builds and runs just fine natively. We've also confirmed that it builds successfully under nix. I think there must be something separate going on with @shinzui and @ixnij's environment that may be causing problems

Please see: https://github.com/ellie/atuin/issues/220

cc @uhthomas and @conradludgate, both of whom have build Atuin on M1 other than myself

**Comment by SuperSandro2000** [comment]
@ixnij do you have the sandbox enabled?

**Comment by ixnij** [comment]
> @ixnij do you have the sandbox enabled?

Yes.

**Comment by SuperSandro2000** [comment]
sandbox on darwin is sometimes broken. It probably tries to do something impure to check something for darwin. We should patch that and hardcode the check result on darwin.

**Pull request merged**